### PR TITLE
separate out Contributing guide from ReadMe.md .

### PR DESCRIPTION
**Pull request by harrieshin**
### Platforms Impacted
- [x] iOS
- [x] macOS

### Description of changes

Also create different contributing.md per platform to give bit more clarity which projects to touch.
Also add list of new controls we have added in Readme.

### Verification

n/a

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/961)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Make sure your local [Xcode version](https://github.com/microsoft/fluentui-apple#requirements) is the same as the supported version of the repo. 
+
+#### Swift Lint
+This project uses [SwiftLint](https://github.com/realm/SwiftLint) to automatically lint our Swift code for common errors. Please install it when developing in this repo by following the [SwiftLint Installation Instructions](https://realm.github.io/SwiftLint/).
+
+### Developing in the repo
+
+Fluent UI Apple requires all [pull requests](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests) to come from forks of the repository. Please see [Fork a Repo - GitHub Help](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) for more details on how to set up a fork of Microsoft/fluentui-apple, keep it up-to-date, and submit pull requests back to this repository.
+
+Fluent UI Apple doesn't have any external code dependencies, so developing in the repository is as easy as launching the appropriate Xcode project or workspace and building and running a test/demo app.
```
Comment (edjamesmsft) [inline CONTRIBUTING.md:28]: Is it worth it to specify what "the appropriate project/workspace" is, or how to run the demo app? If we already have that outlined somewhere, we should link it here.

**Review by mischreiber — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (edjamesmsft) [reply CONTRIBUTING.md:28]: Ah, reading further down, we DO specify. I think we should re-word this to say something like (but better than) "instructions to run the demo app are in the respective CONTRIBUTING.MD files for ios and macos" with links to those files.

File: ios/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- In the Xcode scheme menu choose `Demo.development` and choose an iOS Simulator(or Device if you have your own device provisioning profile) to deploy to.
+- Make sure Swift Package Manager has download 2 dependencies(appcenter-sdk-apple and PLCrashReporter) for demo app.
+- Build and Run the demo app
+
+## Adding a new component
+- Create a new folder of the Control name (ex. Foo) under under [FluentUI](FluentUI)
+- Create a new swift file. (ex. Foo.swift)
+- Add Foo.swift in Fluent xcode project (All the files are in alphabetical order)
+- Make sure your file is under FluentUILib target
+- Create and add a demo controller in the Fluent demo app under [Demos](FluentUI.Demo/FluentUI.Demo/Demos) (ex.FooDemoController.swift)
```
Comment (edjamesmsft) [inline ios/CONTRIBUTING.md:19]: Nit: spacing - "ex.FooDemoController.swift" should be "ex. FooDemoController.swift"

File: ios/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- Add FooDemoController to Fluent Demo xcode project (All the files are in alphabetical order)
```
Comment (edjamesmsft) [inline ios/CONTRIBUTING.md:20]: Nit: "FooDemoController.swift"?

File: ios/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- Make sure its part of FluentUI.Demo Target membership
```
Comment (edjamesmsft) [inline ios/CONTRIBUTING.md:21]: Nit: "it's"

File: ios/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- Add your FooDemoController in list of [DemoDescriptor](https://github.com/microsoft/fluentui-apple/blob/02b1c3fe601b793cb6cfd24813e11d92420e0d77/ios/FluentUI.Demo/FluentUI.Demo/Demos.swift#L30)
```
Comment (edjamesmsft) [inline ios/CONTRIBUTING.md:22]: Nit: "Add your FooDemoController to list of" not "in list of"? Very minor though

File: macos/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Fluent UI for macOS
+
+## Demo app
+
+Included in this repository is a demo app of currently showcase all the available Fluent controls.
```
Comment (edjamesmsft) [inline macos/CONTRIBUTING.md:5]: The previous CONTRIBUTING.md file has this as its first line: "Included in this repository is a demo of currently implemented controls."

Should they be consistent? If not, this one should probably be re-written to be something like "Included in this repository is a demo app showcasing all of the currently available Fluent controls."

File: macos/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+- Once deployed you can choose a control to test from the list of controls in the left pane.
+
+### Adding a new component
+- Create a new folder of the Control name (ex. Foo) under [FluentUI](FluentUI)
+- Create a new swift file. (ex. Foo.swift)
+- Add Foo.swift and its folder in FluentUI xcode project (All the files are in alphabetical order)
+- Make sure your file is under FluentUI framework target
+- Create and add a test ViewController under [FluentUITestViewControllers](FluentUITestViewControllers) (ex. TestFooViewController.swift)
+- Add TestFooViewController.swift to FluentUI xcode project (All the files are in alphabetical order)
+- Make sure its part of FluentUITestViewControllers Target membership
```
Comment (edjamesmsft) [inline macos/CONTRIBUTING.md:20]: Nit: "it's"

File: macos/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+- Add TestFooViewController in the array in [TestViewContronllers.swift](https://github.com/microsoft/fluentui-apple/blob/02b1c3fe601b793cb6cfd24813e11d92420e0d77/macos/FluentUITestViewControllers/TestViewControllers.swift#L14)
```
Comment (edjamesmsft) [inline macos/CONTRIBUTING.md:21]: Nit: "to the array in"

File: macos/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+- Add a new cocopod subspec in [MicrosoftFluentUI.podspec](https://github.com/microsoft/fluentui-apple/blob/main/MicrosoftFluentUI.podspec) with required dependencies. "s.subspec 'Foo_mac' do |foo_mac|"
+- Verify by "pod spec lint" For more info on [cocoapod](https://cocoapods.org)
+- Build and Run
+- Make sure no warnings and errors. Test your components!
+- Add documentation for your new class and especially for public apis
```
Comment (edjamesmsft) [inline macos/CONTRIBUTING.md:26]: For iOS, you recommend adding unit tests explicitly. Should that instruction be the same here?

**Review by edjamesmsft — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (harrieshin) [reply CONTRIBUTING.md:28]: hmmm i don't want it to be too wordy but i reword the next sentence a bit
